### PR TITLE
[PostRector][BC Break] Remove CurrentFileProvider usage on UseAddingPostRector and UseNodesToAddCollector

### PR DESCRIPTION
**Pull request by samsonasik**
@TomasVotruba here based on discussion on https://github.com/rectorphp/rector-src/pull/6049#pullrequestreview-2142272639

### CODE REVIEW

**Comment by samsonasik** [comment]
Tested on rector-src code base, got 1 seconds faster, it because there everyting already imported, so nothing to append:

**Before** **36.474** seconds

```
➜  rector-src git:(main) time bin/rector process --clear-cache
 2230/2230 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

                                                                                                                        
 [OK] Rector is done!                                                                                                   
                                                                                                                        
bin/rector process --clear-cache  255.91s user 10.41s system 730% cpu 36.474 total
```

**After** **35.850** seconds

```
➜  rector-src git:(remove-current-file-provider) time bin/rector process --clear-cache    
 2230/2230 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%
                                                                                                                        
 [OK] Rector is done!                                                                                                   
                                                                                                                        
bin/rector process --clear-cache  252.20s user 9.94s system 731% cpu 35.850 total
```

Basically, this PR remove append by index `filePath`, instead, on next file, it reset to empty data instead.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: rules/CodingStyle/ClassNameImport/ClassNameImportSkipVoter/UsesClassNameImportSkipVoter.php [hunk 1]
```diff
@@ -26,7 +26,7 @@ public function __construct(
 
     public function shouldSkip(File $file, FullyQualifiedObjectType $fullyQualifiedObjectType, Node $node): bool
     {
-        $useImportTypes = $this->useNodesToAddCollector->getUseImportTypesByNode($file);
+        $useImportTypes = $this->useNodesToAddCollector->getUseImportTypes();
```
Comment (TomasVotruba) [inline rules/CodingStyle/ClassNameImport/ClassNameImportSkipVoter/UsesClassNameImportSkipVoter.php:29]: It's not clear what the method returns now.  Probably depends on some inner order, which is not good idea.

**Comment by TomasVotruba** [comment]
Seems there is no difference in performance, but method API is now unclear.
Don't worry about this, I'll deep dive and explore it myself :)

**Comment by samsonasik** [comment]
The no different performance is because there is no file diff, as everything already imported in rector-src, to test this exactly, the big project with not have auto imported everywhere can be usage for test.

**Comment by TomasVotruba** [comment]
The goal here is not the performance, but to remove this setter/getter service. 
I made this bad design choice and now it's unclear what file is where :)

**Pull request closed without merging**